workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -162,32 +162,32 @@ namespace <#= qualifiedNamespace #>
                 if (entityManager.HasComponent<NotAuthoritative<<#= componentNamespace #>.Component>>(entity))
                 {
                     var data = entityManager.GetComponentData<<#= componentNamespace #>.Component>(entity);
+                    <#= componentNamespace #>.Serialization.ApplyUpdate(op.Update.SchemaData.Value.GetFields(), ref data);
+                    data.DirtyBit = false;
+                    entityManager.SetComponentData(entity, data);
+                }
 
-                    var update = <#= componentNamespace #>.Serialization.GetAndApplyUpdate(op.Update.SchemaData.Value.GetFields(), ref data);
+                var update = <#= componentNamespace #>.Serialization.DeserializeUpdate(op.Update.SchemaData.Value.GetFields());
 
-                    List<<#= componentNamespace #>.Update> updates;
-                    if (entityManager.HasComponent<<#= componentNamespace #>.ReceivedUpdates>(entity))
-                    {
-                        updates = entityManager.GetComponentData<<#= componentNamespace #>.ReceivedUpdates>(entity).Updates;
+                List<<#= componentNamespace #>.Update> updates;
+                if (entityManager.HasComponent<<#= componentNamespace #>.ReceivedUpdates>(entity))
+                {
+                    updates = entityManager.GetComponentData<<#= componentNamespace #>.ReceivedUpdates>(entity).Updates;
 
-                    }
-                    else
+                }
+                else
+                {
+                    var updatesComponent = new <#= componentNamespace #>.ReceivedUpdates
                     {
-                        var updatesComponent = new <#= componentNamespace #>.ReceivedUpdates
-                        {
-                            handle = ReferenceTypeProviders.UpdatesProvider.Allocate(World)
-                        };
-                        ReferenceTypeProviders.UpdatesProvider.Set(updatesComponent.handle, new List<<#= componentNamespace #>.Update>());
-                        updates = updatesComponent.Updates;
-                        entityManager.AddComponentData(entity, updatesComponent);
-                    }
-
-                    updates.Add(update);
-
-                    data.DirtyBit = false;
```

Comment:
should the dirtybit still get affected here?

Reply:
I guess there is no `data` left so sounds like not

Reply:
You are correct!